python/ray/serve/backend_state.py

```diff
@@ -260,30 +373,38 @@ def _start_backend_replica(self, backend_tag: BackendTag,
     def scale_backend_replicas(
             self,
             backend_tag: BackendTag,
-            num_replicas: int,
             force_kill: bool = False,
     ) -> None:
         """Scale the given backend to the number of replicas.
 
         NOTE: this does not actually start or stop the replicas, but instead
-        adds the intention to start/stop them to self.backend_replicas_to_start
-        and self.backend_replicas_to_stop. The caller is responsible for then
-        first writing a checkpoint and then actually starting/stopping the
-        intended replicas. This avoids inconsistencies with starting/stopping a
-        replica and then crashing before writing a checkpoint.
+        adds them to ReplicaState.SHOULD_START or ReplicaState.SHOULD_STOP.
+        The caller is responsible for then first writing a checkpoint and then
+        actually starting/stopping the intended replicas. This avoids
+        inconsistencies with starting/stopping a replica and then crashing
+        before writing a checkpoint.
         """
+        num_replicas = self._target_replicas.get(backend_tag, 0)
```

Comment:
why do we need the default to zero here? we should always have a target replica count if the backend exists, right?

Reply:
Correct!